src/scanner/mod.rs

```diff
@@ -87,54 +98,71 @@ impl Scanner {
-                    &socket
-                );
-                match x.shutdown(Shutdown::Both) {
-                    Err(e) => debug!("Shutdown stream error {}", &e),
-                    _ => {}
-                }
-                if !self.greppable {
-                    if self.accessible {
-                        println!("Open {}", socket.to_string());
-                    } else {
-                        println!("Open {}", socket.to_string().purple());
+        let tries = self.tries.get();
+
+        debug!("self.tries: {}", tries);
+
+        for nr_try in 1..=tries {
+            debug!("Try number: {}", nr_try);
+
+            match self.connect(socket).await {
+                Ok(x) => {
+                    debug!(
+                        "Connection was successful, shutting down stream {}",
+                        &socket
+                    );
+                    match x.shutdown(Shutdown::Both) {
+                        Err(e) => debug!("Shutdown stream error {}", &e),
+                        _ => {}
                     }
+                    if !self.greppable {
+                        if self.accessible {
+                            println!("Open {}", socket.to_string());
+                        } else {
+                            println!("Open {}", socket.to_string().purple());
+                        }
+                    }
+
+                    debug!("Return Ok after {} tries", nr_try);
+                    return Ok(socket);
                 }
+                Err(e) => {
+                    let error_string = e.to_string();
 
-                Ok(socket)
-            }
-            Err(e) => match e.kind() {
-                ErrorKind::Other => {
-                    if e.to_string().contains("No route to host")
-                        || e.to_string().contains("Network is unreachable")
-                    {
-                        debug!("Socket connect error: {} {}", &e.to_string(), &socket);
-                        Err(io::Error::new(io::ErrorKind::Other, e.to_string()))
-                    } else {
-                        debug!("Socket connect error: {} {}", &e.to_string(), &socket);
-                        panic!("Too many open files. Please reduce batch size. The default is 5000. Try -b 2500.");
+                    if e.kind() == ErrorKind::Other {
+                        debug!("Socket connect error: {} {}", &error_string, &socket);
+
+                        if !error_string.contains("No route to host")
+                            && !error_string.contains("Network is unreachable")
+                        {
+                            panic!("Too many open files. Please reduce batch size. The default is 5000. Try -b 2500.");
+                        }
+                    }
+
+                    if nr_try == tries {
+                        debug!("Return Err after {} tries", tries);
+                        return Err(io::Error::new(io::ErrorKind::Other, e.to_string()));
```

Comment:
You can use `error_string` here

Reply:
For the `unwrap_or`, see my [above comment](https://github.com/RustScan/RustScan/pull/266#discussion_r505733950).

For the `error_string`, next time I use the regex of my editor to replace them all!